src/shared/sleep-config.c

```diff
@@ -108,77 +111,132 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+/* Get the list of battery */
+static int get_sd_device_battery(sd_device_enumerator *e) {
```

Comment:
To initialize a pointer in a function, you need to pass the address of the pointer to the function, so this should be `sd_device_enumerator **e`

Reply:
also, call it `ret`, and use a local _cleanup_ sd_device_enumerator to do the initialization, and TAKE_PTR at the end

Reply:
Fixed.